Source/VesselData.cs

```diff
@@ -1,798 +1,890 @@
+                        if (Persistence == HighLogic.SaveFolder.ToString() || Persistence == "WhitecatsDummySaveFileThatNoOneShouldNameTheirSave")
+                        {
+                            VesselInformation.AddNode(vessel);
+                        }
+                    }
+                }
+            }
+            print("WhitecatIndustries - Orbital Decay - Loaded vessel data.");
+        }
+
+        public void FixedUpdate()
+        {
+            if (Time.timeSinceLevelLoad > 0.3)
+            {
+                if ((Time.time - lastUpdate) > UPTInterval) // 1.4.0 Lag Busting
+                {
+                    lastUpdate = Time.time;
+
+                    if (HighLogic.LoadedSceneIsGame && (HighLogic.LoadedScene != GameScenes.LOADING && HighLogic.LoadedScene != GameScenes.LOADINGBUFFER && HighLogic.LoadedScene != GameScenes.MAINMENU))
+                    {
+                        Vessel vessel = new Vessel();
+                        for (int i = 0; i < FlightGlobals.Vessels.Count; i++)
+                        {
+                            vessel = FlightGlobals.Vessels.ElementAt(i);
+                            {
+                                if (CheckIfContained(vessel) == true)
+                                {
+                                    if (vessel.situation != Vessel.Situations.SUB_ORBITAL && vessel.situation != Vessel.Situations.ORBITING)
+                                    {
+                                        ClearVesselData(vessel);
+                                    }
+
+                                    if (vessel.situation == Vessel.Situations.ORBITING || vessel.situation == Vessel.Situations.SUB_ORBITAL)
+                                    {
+                                        WriteVesselData(vessel);
+                                    }
+                                }
+                                else if (CheckIfContained(vessel) == false)
+                                {
+                                    if (vessel.situation == Vessel.Situations.ORBITING || vessel.situation == Vessel.Situations.SUB_ORBITAL) // 1.4.2
+                                    {
+                                        WriteVesselData(vessel);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public void OnDestroy()
+        {
+            if (DecayManager.CheckSceneStateMain(HighLogic.LoadedScene))
+            {
+                if ((Planetarium.GetUniversalTime() == HighLogic.CurrentGame.UniversalTime) || HighLogic.LoadedScene == GameScenes.FLIGHT)
+                {
+                    print("WhitecatIndustries - Orbital Decay - Vessel Information saved. Ondestroy");
+                    File.ClearNodes();
+                   // VesselInformation.Save(FilePath);

```

Comment:
New code does use fuel and resources field, but really only to minimise changes to rest of the code. it can be removed later on, but changes to VesselData.FetchFuel() and few more methods  will be needed. Reading directly from protovessel might proove to be causing lag issues though.